apis/apim/v1alpha1/common.go

```diff
@@ -0,0 +1,50 @@
+package v1alpha1
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/google/go-cmp/cmp"
+	"github.com/kuadrant/kuadrant-controller/pkg/common"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Interface to be implemented by Policy's status struct
+//+kubebuilder:object:generate=false
+type PolicyStatus interface {
+	GetObservedGeneration() int64
+	GetConditions() []metav1.Condition
+}
+
+func (r RateLimitPolicyStatus) GetObservedGeneration() int64 {
```

Comment:
I think it can be either and only how you use it varies with it. If required in the future for a pointer receiver then no problem in changing it to that :)